instrumentation/net/http/otelhttp/transport.go

```diff
@@ -109,7 +109,7 @@ func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
 		ctx = httptrace.WithClientTrace(ctx, t.clientTrace(ctx))
 	}
 
-	r = r.WithContext(ctx)
+	r = r.Clone(ctx) // According to RoundTripper spec we shouldn't modify the origin request.
```

Comment:
Can a unit test be added to verify that the original request is not modified so that we can avoid regression in the future?